game/script-farewells.rpy

```diff
@@ -2,42 +2,1223 @@ default persistent._farewell_database = dict()
 
 init python in farewells:
     import random
+    import store
+
     FAREWELL_MAP = dict()
 
     def select_farewell():
         """
-        Picks a random greeting
+        Picks a random farewell, accounting for affinity
+        If the player has already been asked to stay by Natsuki, a farewell without the option
+        to stay will be selected
+        """
+        # Get the farewells the current affinity allows for the player
+        #TODO: Generalized filter function
+        farewell_pool = filter(lambda x: x.evaluate_affinity_range(), FAREWELL_MAP.values())
+
+        # Filter any time-sensitive topics, if the player has been around 30 minutes or longer
+        if divmod((store.utils.get_current_session_length()).total_seconds(), 60)[0] > 30:
```

Comment:
Same divmod here

Reply:
This divmod can go, since the comparison doesn't really depend on an integer.